src/Octoshift/AzureApi.cs

```diff
@@ -57,6 +58,26 @@ public virtual async Task<Uri> UploadToBlob(string fileName, byte[] content)
             return GetServiceSasUriForBlob(blobClient);
         }
 
+        public virtual async Task<Uri> UploadToBlob(string fileName, Stream content)
+        {
+            var containerClient = await CreateBlobContainerAsync();
+            var blobClient = containerClient.GetBlobClient(fileName);
+
+            var options = new BlobUploadOptions
+            {
+                TransferOptions = new Azure.Storage.StorageTransferOptions()
+                {
+                    InitialTransferSize = 4 * 1024 * 1024,
+                    MaximumTransferSize = 4 * 1024 * 1024
+                },
+            };
+
+            content.Position = 0;
+            await blobClient.UploadAsync(content, options);
+            content.Dispose();
```

Comment:
same here too. disposing the stream is not this method's responsibility.

Reply:
Addressed in 82ac9f5d78b852715173f9948e0b7b7a712522bb